File: sgps-api/src/main/java/br/com/henrique/sgps/domain/Participante.java
package br.com.henrique.sgps.domain;

import br.com.henrique.sgps.domain.enuns.Classe;
import br.com.henrique.sgps.domain.enuns.Nivel;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Participante implements Serializable {
    @Serial
    private static final long serialVersionUID = -9134120161347242146L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column
    private String nome;
    @Column
    private String cpf;
    @Column(name = "dt_nascimento")
    private LocalDate dataNascimento;
    @Column(name = "dt_ingresso")
    private LocalDate dataIngresso;
    @Column
    @Enumerated(EnumType.STRING)
    private Classe classe;
    @Column

    private Nivel nivel;
    @OneToOne
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    @OneToMany(mappedBy = "participante")
    private Set<Inscricao> inscricoes;

    @Transient
    public Double getPontos() {
        Double pontosClasse = this.getClasse().getPeso();
        Double pontosNivel = this.getNivel().getPeso();

        return pontosClasse + pontosNivel;
    }
}

File: sgps-app/lib/domain/models/participante.dart
import 'package:sgps/domain/models/incricao.dart';

class Participante {
  int? id;
  String? nome;
  String? cpf;
  String? confirmacaoCpf;
  String? dataNascimento;
  String? dataIngresso;
  String? classe;
  String? nivel;
  String? senha;
  String? confirmacaoSenha;
  int? idProcessoSeletivo;
  List<Inscricao>? inscricoes;

  Participante(
      {this.id,
      this.nome,
      this.cpf,
      this.confirmacaoCpf,
      this.dataNascimento,
      this.dataIngresso,
      this.classe,
      this.nivel,
      this.senha,
      this.confirmacaoSenha,
      this.idProcessoSeletivo,
      this.inscricoes});

  Participante.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    cpf = json['cpf'];
    confirmacaoCpf = json['confirmacaoCpf'];
    dataNascimento = json['dataNascimento'];
    dataIngresso = json['dataIngresso'];
    classe = json['classe'];
    nivel = json['nivel'];
    senha = json['senha'];
    confirmacaoSenha = json['confirmacaoSenha'];
    idProcessoSeletivo = json['idProcessoSeletivo'];
    if (json['inscricaos'] != null) {
      inscricoes = <Inscricao>[];
      json['inscricaos'].forEach((v) {
        inscricoes!.add(Inscricao.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['nome'] = nome;
    data['cpf'] = cpf;
    data['confirmacaoCpf'] = confirmacaoCpf;
    data['dataNascimento'] = dataNascimento;
    data['dataIngresso'] = dataIngresso;
    data['classe'] = classe;
    data['nivel'] = nivel;
    data['senha'] = senha;
    data['confirmacaoSenha'] = confirmacaoSenha;
    data['idProcessoSeletivo'] = idProcessoSeletivo;
    if (inscricoes != null) {
      data['inscricaos'] = inscricoes!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}
